src/hivpy/hiv_testing.py

```diff
@@ -84,29 +84,70 @@ def update_hiv_testing(self, pop):
                 # set outcomes
-                pop.set_present_variable(col.EVER_TESTED, tested, sub_pop=untested_population)
-                # set last test date
-                pop.set_present_variable(col.LAST_TEST_DATE, pop.date,
-                                         sub_pop=pop.apply_bool_mask(tested, untested_population))
-                # "reset" dummy partner columns
-                pop.set_present_variable(col.NSTP_LAST_TEST, 0,
-                                         sub_pop=pop.apply_bool_mask(tested, untested_population))
-                pop.set_present_variable(col.NP_LAST_TEST, 0,
-                                         sub_pop=pop.apply_bool_mask(tested, untested_population))
+                pop.set_present_variable(col.EVER_TESTED, tested, untested_population)
+                self.apply_test_outcomes_to_sub_pop(pop, tested, untested_population)
 
             if len(prev_tested_population) > 0:
                 # test repeat testers
                 tested = pop.transform_group([col.EVER_TESTED, col.NP_LAST_TEST, col.NSTP_LAST_TEST],
                                              self.calc_testing_outcomes,
                                              sub_pop=prev_tested_population)
-                # set last test date
+                self.apply_test_outcomes_to_sub_pop(pop, tested, prev_tested_population)
+
+    def apply_test_outcomes_to_sub_pop(self, pop, tested, sub_pop):
+        """
+        Uses HIV testing outcomes for a given sub-population to
+        set last test date and reset number of partners since last test.
+        """
+        # set last test date
+        pop.set_present_variable(col.LAST_TEST_DATE, pop.date,
+                                 sub_pop=pop.apply_bool_mask(tested, sub_pop))
+        # "reset" dummy partner columns
+        pop.set_present_variable(col.NSTP_LAST_TEST, 0,
+                                 sub_pop=pop.apply_bool_mask(tested, sub_pop))
+        pop.set_present_variable(col.NP_LAST_TEST, 0,
+                                 sub_pop=pop.apply_bool_mask(tested, sub_pop))
+
+    def update_anc_hiv_testing(self, pop):
+        """
+        Update which pregnant women are tested while in antenatal care.
+        COVID disruption is factored in.
+        """
+        # conduct two tests in anc during pregnancy if there is no covid disruption
+        if not (self.covid_disrup_affected | self.testing_disrup_covid):
+            # get population in second trimester
+            in_second_trimester = pop.get_sub_pop([(col.HIV_STATUS, op.eq, False),
+                                                   (col.ANC, op.eq, True),
+                                                   (col.PREGNANT, op.eq, True),
+                                                   (col.LAST_PREGNANCY_DATE, op.lt, pop.date - timedelta(days=90)),
+                                                   (col.LAST_PREGNANCY_DATE, op.ge, pop.date - timedelta(days=180))])
```

Comment:
```suggestion
                                                   (col.LAST_PREGNANCY_DATE, op.lt, pop.date - timedelta(days=90)),
                                                   (col.LAST_PREGNANCY_DATE, op.ge, pop.date - time_delta(days=90) + time_step)])
```

Given the SAS code this should only execute once during pregnancy. In the SAS it happens exactly 3 months into pregnancy, but because we are being timestep agnostic we can make it just the first time step which is at least 3 months into pregnancy.